src/clients/chunker.rs

```diff
@@ -53,15 +52,29 @@ type StreamingTokenizationResult =
 #[cfg_attr(test, faux::create)]
 #[derive(Clone)]
 pub struct ChunkerClient {
-    client: ChunkersServiceClient<LoadBalancedChannel>,
-    health_client: HealthClient<LoadBalancedChannel>,
+    client: GrpcClient<ChunkersServiceClient<LoadBalancedChannel>>,
+    health_client: GrpcClient<HealthClient<LoadBalancedChannel>>,
 }
 
 #[cfg_attr(test, faux::methods)]
 impl ChunkerClient {
     pub async fn new(config: &ServiceConfig) -> Self {
-        let client = create_grpc_client(DEFAULT_PORT, config, ChunkersServiceClient::new).await;
-        let health_client = create_grpc_client(DEFAULT_PORT, config, HealthClient::new).await;
+        let client = create_grpc_client(
```

Comment:
Explanation:

because of prost autogenerated protobuf, information that would be runtime string information in HTTP (eg url passed to client), is static type information in grpc, making tracing certain details as string fields added to spans difficult, since rust does not support dynamic type reflection. I still dont have a great solution for getting a given grpc request's service method/endpoint in the middleware, but we can at least hold on to the service name with this method.

Also: the new boolean as the last arg to `create_grpc_client` is for enabling tracing. The thought process being that health clients typically dont want to have tracing enabled (causes middleware to skip collecting metrics on every request, allows health requests to run faster/frequently)